core/elem.ts

```diff
@@ -65,13 +65,11 @@ export type ElementSpec<P> = P & {
      * Hoist components accept and should use.
      */
     [TEST_ID]?: string;
-
+} & {
     //----------------------------
     // Technical -- Escape support
     //----------------------------
-    $items?: any;
-    $item?: any;
-    $omit?: any;
+    [K in keyof P & ('items' | 'item' | 'omit') as `$${K}`]?: P[K];
```

Comment:
medicine worse than the cure here?

Reply:
Hmm not sure. Nice to have the types right for these "reserved" props for components that use them, though we don't expect that to happen often.